packages/flutter/lib/src/widgets/navigation_toolbar.dart

```diff
@@ -0,0 +1,136 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/rendering.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+
+/// [NavigationToolbar] is an unthemed layout helper to position 3 groups
+/// of widgets along a horizontal axis that's sensible for an application's
+/// navigation bar such as in Material Design and in iOS.
+///
+/// Optional [leading] and [trailing] widgets occupy the edges of the widget
+/// with reasonable constraints while the [middle] widget occupies the remaining
+/// space in either a center aligned or start aligned fashion.
+///
+/// Either directly use the themed app bars such as the Material [AppBar] or
+/// the iOS [CupertinoNavigationBar] or wrap this widget with more theming
+/// specifications for your own custom app bar.
+class NavigationToolbar extends StatelessWidget {
```

Comment:
I wish this had a more generic name, since it's just a layout widget, albeit a somewhat specialized one. That said, I don't have any great ideas (ThreeBar? Ugh).